Author: Pino Toscano <pino@kde.org>
Description: preliminary GNU/Hurd support
 Adds "support" for GNU/Hurd by defining MAXPATHLEN and PATH_MAX where needed.
 Not forwarded, better fixes should be done upstream.
Forwarded: no
Index: kde4libs/kdecore/kernel/kstandarddirs.cpp
===================================================================
--- kde4libs.orig/kdecore/kernel/kstandarddirs.cpp	2010-11-25 16:40:12.000000000 +0100
+++ kde4libs/kdecore/kernel/kstandarddirs.cpp	2010-11-25 16:40:13.000000000 +0100
@@ -73,6 +73,10 @@
 #include <QtCore/QFileInfo>
 #include <QtCore/QSettings>
 
+#ifndef MAXPATHLEN
+# define MAXPATHLEN 1024
+#endif
+
 class KStandardDirs::KStandardDirsPrivate
 {
 public:
Index: kde4libs/kinit/lnusertemp.c
===================================================================
--- kde4libs.orig/kinit/lnusertemp.c	2008-06-18 14:42:25.000000000 +0200
+++ kde4libs/kinit/lnusertemp.c	2010-11-25 16:40:13.000000000 +0100
@@ -36,6 +36,10 @@
 #include <limits.h>
 #endif
 
+#ifndef PATH_MAX
+# define PATH_MAX 1024
+#endif
+
 int check_tmp_dir(const char *tmp_dir, int check_ownership);
 int create_link(const char *file, const char *tmp_dir);
 int build_link(const char* tmp, const char *tmp_prefix, const char *kde_prefix);
